src/typing/flow_js.ml

```diff
@@ -6558,6 +6558,18 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       (GetPropT _ | SetPropT _ | MethodT _ | LookupT _) ->
       rec_flow cx trace (get_builtin_typeapp cx ~trace reason "Array" [t], u)
 
+    (*************************)
+    (* Tuple "length" access *)
+    (*************************)
+
+    | DefT (_, trust, ArrT (TupleAT (_, ts))),
+      LookupT (reason, _, _, Named (_, "length"), ReadProp {
```

Comment:
Can you add `GetPropT` to right hand match?

Reply:
I can definitely do it, but at the moment I do not understand what it would mean.
I will spend some time reading the code, to understand why would it be better.

Reply:
Currently `x.length` doesn't work on tuples outside of `if`, it is obvious if you change number to number literal in `tuple_length`

Reply:
Replaced `LookupT` with `GetPropT`.

Reply:
And then replaced back with both versions :)